app/models/accounting/quickbooks/transaction_extractor.rb

```diff
@@ -17,86 +43,63 @@ def extract!
             txn.line_items.destroy(li) unless qb_ids.include?(li.qb_line_id)
           end
         end
+      end
 
+      def extract_line_items
         txn.quickbooks_data['line_items'].each do |li|
-          acct = Accounting::Account.find_by(qb_id: li['journal_entry_line_detail']['account_ref']['value'])
+          acct = Accounting::Account.find_by(qb_id: li[qb_li_detail_key]['account_ref']['value'])
 
           # skip if line item does not have an account in Madeline
           next unless acct
 
           txn.line_item_with_id(li['id'].to_i).assign_attributes(
             account: acct,
             amount: li['amount'],
-            posting_type: li['journal_entry_line_detail']['posting_type']
+            posting_type: li[qb_li_detail_key]['posting_type']
           )
         end
```

Comment:
Is now a good time to combine this logic (for DRY) or should we wait and see how the other subclasses shake out?